src/modules/mc_att_control/mc_att_control_main.cpp

```diff
@@ -848,77 +850,26 @@ void
 MulticopterAttitudeControl::control_attitude(float dt)
 {
 	vehicle_attitude_setpoint_poll();
-
 	_thrust_sp = _v_att_sp.thrust;
 
-	/* construct attitude setpoint rotation matrix */
-	math::Quaternion q_sp(_v_att_sp.q_d[0], _v_att_sp.q_d[1], _v_att_sp.q_d[2], _v_att_sp.q_d[3]);
-	math::Matrix<3, 3> R_sp = q_sp.to_dcm();
-
-	/* get current rotation matrix from control state quaternions */
-	math::Quaternion q_att(_v_att.q[0], _v_att.q[1], _v_att.q[2], _v_att.q[3]);
-	math::Matrix<3, 3> R = q_att.to_dcm();
-
-	/* all input data is ready, run controller itself */
-
-	/* try to move thrust vector shortest way, because yaw response is slower than roll/pitch */
-	math::Vector<3> R_z(R(0, 2), R(1, 2), R(2, 2));
-	math::Vector<3> R_sp_z(R_sp(0, 2), R_sp(1, 2), R_sp(2, 2));
-
-	/* axis and sin(angle) of desired rotation */
-	math::Vector<3> e_R = R.transposed() * (R_z % R_sp_z);
-
-	/* calculate angle error */
-	float e_R_z_sin = e_R.length();
-	float e_R_z_cos = R_z * R_sp_z;
-
-	/* calculate weight for yaw control */
-	float yaw_w = R_sp(2, 2) * R_sp(2, 2);
+	using namespace matrix;
+	float yaw_w = .4f;
```

Comment:
Constant 0.4?

Reply:
Yes, I can recommend you to read section 4.2 on page 16 of the paper I linked to about what this weight actually should represent and some comparison of resulting behaviours. I personally found 0.4 a good rule of thumb for a typical quadrotor setup in my past experience, that's why I put it there for the time being. I think this should be incorporated with the attitude yaw gain parameter in the end. Still to do.

It would be interesting where the current yaw weight calculation comes from and how it's justified from a control perspective. @bkueng told me already that it's done wrong and has his improvement suggestion here: https://github.com/PX4/Firmware/pull/7997/files#diff-81c4f50bdf837c4183b5d3cb81eb40eaR898